tutorials/tensordictmodule.ipynb

```diff
@@ -0,0 +1,1075 @@
+    "    *[\n",
+    "        splitlinear.make_functional_with_buffers()[1]\n",
+    "        for splitlinear in splitlinear_models\n",
+    "    ]\n",
+    ")\n",
+    "## For simplicity we reinit the params. In a real application you need to stack all params\n",
+    "params = [torch.randn(num_models, *p.shape) for p in params[0]]\n",
+    "buffers = [torch.randn(num_models, *b.shape) for b in buffers[0]]\n",
+    "\n",
+    "### Finally we can apply the function to the stack of params and buffer using vmap\n",
+    "func(tensordict, params=params, buffers=buffers, vmap=True).shape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "31be6c45-10fb-4fd1-a52f-92214b76c00a",
+   "metadata": {},
+   "source": [
+    "## Do's and don't with `TensorDictModule`\n",
+    "\n",
+    "Don't use `nn.Module` wrappers with `TensorDictModule` componants. This would break some of `TensorDictModule` features such as `functorch` compatibility. \n",
+    "\n",
+    "Don't use `nn.Sequence`, similar to nn.Module, it would break features such as `functorch` compatibility.\n",
+    "\n",
+    "Do use `TensorDictSequence` instead.\n",
+    "\n",
+    "Don't use a different name for the output of a `TensorDictModule` as the output tensordict is just the input modified in-place:\n",
+    "\n",
+    "```python\n",
+    "tensordict = module(tensordict)  # ok!\n",
+    "tensordict_out = module(tensordict)  # don't!\n",
+    "```\n",
+    "\n",
+    "Don't use `make_functional_with_buffers` from `functorch` directly.\n",
+    "\n",
+    "Do use `TensorDictModule.make_functional_with_buffers` instead.\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "22e65356-d8b3-4197-84b8-598330c1ddc8",
+   "metadata": {},
+   "source": [
+    "## TensorDictModule for RL"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8d49a911-933c-476f-8c9a-00e006ed043c",
+   "metadata": {},
+   "source": [
+    "In the context of RL torchrl offers a few wrappers on `TensorDictModule`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e33904a6-d405-45db-a713-47493ca8ee33",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "### `ProbabilisticTensorDictModule`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fea4eead-47b4-4029-a8ff-e3c3faf51b0f",
+   "metadata": {},
+   "source": [
+    "`ProbabilisticTDModule` is a special case of a `TensorDictModule` where the output is\n",
+    "sampled given some rule, specified by the input `default_interaction_mode`\n",
+    "argument and the `exploration_mode()` global function.\n",
+    "\n",
+    "It consists in a wrapper around another `TensorDictModule` that returns a tensordict\n",
+    "updated with the distribution parameters. `ProbabilisticTensorDictModule` is\n",
+    "responsible for constructing the distribution (through the `get_dist()` method)\n",
+    "and/or sampling from this distribution (through a regular `__call__()` to the\n",
+    "module)."
```

Comment:
we should also say that one can find the parameters in the output tensordict as well as the log probability if needed